cypress/utils/mockErrorResponse.ts

```diff
@@ -0,0 +1,34 @@
+interface Args {
+  errorMessage: string;
+  path?: string;
+  errorCode?:
+    | "INTERNAL_SERVER_ERROR"
+    | "RESOURCE_NOT_FOUND"
+    | "FORBIDDEN"
+    | "INPUT_VALIDATION_ERROR";
+}
+
+// mockErrorResponse returns a graphql query with an error message.
+// Use this to test error responses from queries and mutations.
+// Call mockErrorResponse before performing action that dispatches XHR request.
+export const mockErrorResponse = ({
```

Comment:
Im pretty apposed to mocking data including error responses for e2e tests. Since its not an accurate representation of how a real life system would behave (The error message could change or some other variable could affect the result and the test would not catch it).
 It might be better to mock it in a unit test or integration test if the goal of this test is to just see if an error banner appears.

Reply:
what is the difference between mocking in a unit test or mocking in a cypress test? mocking in a unit or integration tests just means more tests for the same component spread out across different files. and it's not just about showing a banner and the error message is irrelevant. it's about testing that when the gql response errors out that the app handles it gracefully and displays what it's supposed to.

Reply:
It might be just the connotation of this being an e2e test and not an integration test. Its fine if we do this in a cypress test since that is just the tool. But i think its important to have that distinction between the two. Having more tests for the same component isn't necessarily bad since we are testing different functionalities for the same component. Always mocking the error response from a query or mutation wont allow us to catch a scenario where the behavior  of that query or mutation changes. If something changes on the backend and that query no longer returns an error we wont catch it in a test.

Reply:
1. not all errors are mocked, i.e. requesting a patch ID that does not exist.
2. all queries will always have the potential of returning an error. what that errors says or its code does not usually matter. the important piece of the test is testing that application handles that error. 
3. if the same capabilities as an integration test can be achieved via Cypress e2e tests then we can and should use those capabilities. Using an integration test just to use an integration test is not a valid pro.